torchaudio/prototype/models/hdemucs.py

```diff
@@ -0,0 +1,939 @@
+            channels (int): initial number of hidden channels. (Default: 48)
+            channels_time (Optional[int]): if not None, use a different `channels` value for the time branch.
+                (Default: ``None``)
+            growth (int): increase the number of hidden channels by this factor at each layer. (Default: 2)
+            quality (Optional[int]): determining nfft bins and depth which align with one another.
+                If 0, low quality, if 1, medium quality, if 2, high quality. (Default: ``None``)
+            freq_emb (float): add frequency embedding after the first frequency layer if > 0,
+                the actual value controls the weight of the embedding. (Default: 0.2)
+            emb_scale (int): equivalent to scaling the embedding learning rate (Default: 10)
+            emb_smooth (bool): initialize the embedding with a smooth one (with respect to frequencies).
+                (Default: ``True``)
+            kernel_size (int): kernel_size for encoder and decoder layers. (Default: 8)
+            time_stride (int): stride for the final time layer, after the merge. (Default: 2)
+            stride (int): stride for encoder and decoder layers. (Default: 4)
+            context (int): context for 1x1 conv in the decoder. (Default: 4)
+            context_enc (int): context for 1x1 conv in the encoder. (Default: 0)
+            norm_starts (int): layer at which group norm starts being used.
+                decoder layers are numbered in reverse order. (Default: 4)
+            norm_groups (int): number of groups for group norm. (Default: 4)
+            dconv_depth (int): depth of residual DConv branch. (Default: 2)
+            dconv_comp (int): compression of DConv branch. (Default: 4)
+            dconv_attn (int): adds attention layers in DConv branch starting at this layer. (Default: 4)
+            dconv_lstm (int): adds a LSTM layer in DConv branch starting at this layer. (Default: 4)
+            dconv_init (float): initial scale for the DConv branch LayerScale. (Default: 1e-4)
+            samplerate (int): sample rate, serving as metadata not actually used (Default: 44100)
+            segment (int): segment size (Default: 40)
+        """
+        super().__init__()
+        if quality is None or quality == 2:
+            self.depth = 6
+            self.nfft = 4096
+        elif quality == 1:
+            self.depth = 6
+            self.nfft = 2048
+        else:
+            self.depth = 5
+            self.nfft = 1024
+        self.audio_channels = audio_channels
+        self.sources = sources
+        self.kernel_size = kernel_size
+        self.context = context
+        self.stride = stride
+        # self.depth = depth
+        self.channels = channels
+        self.samplerate = samplerate
+        self.segment = segment
+
+        self.hop_length = self.nfft // 4
+        self.freq_emb = None
+
+        self.encoder = nn.ModuleList()
+        self.decoder = nn.ModuleList()
```

Comment:
I feel like renaming them to `frequency_encoder`, `frequency_decoder` could be more intuitive. what do you think?